WikibaseEnitiyStatementReomover.php

```diff
@@ -0,0 +1,147 @@
+use Wikibase\Api\WikibaseFactory;
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * @author Addshore
+ * @author Lucie-Aimée Kaffee
+ */
+class WikidataEntityStatementRemover extends Command {
+
+	private $appConfig;
+
+	/**
+	 * @var WikibaseFactory
+	 */
+	private $wikibaseFactory;
+
+	/**
+	 * @var MediawikiApi
+	 */
+	private $wikibaseApi;
+
+	/**
+	 * @var SparqlQueryRunner
+	 */
+	private $sparqlQueryRunner;
+
+	public function __construct( ArrayAccess $appConfig ) {
+		$this->appConfig = $appConfig;
+
+		$defaultGuzzleConf = array(
+			'headers' => array( 'User-Agent' => 'addwiki - Wikibase Reference As Statement Remover' )
+		);
+		$guzzleClient = new Client( $defaultGuzzleConf );
+		$this->sparqlQueryRunner = new SparqlQueryRunner( $guzzleClient );
+
+		$this->wikibaseApi = new MediawikiApi( "https://www.wikidata.org/w/api.php" );
+		$this->wikibaseFactory = new WikibaseFactory(
+			$this->wikibaseApi,
+			new DataValueDeserializer(
+				array(
+					'boolean' => 'DataValues\BooleanValue',
+					'number' => 'DataValues\NumberValue',
+					'string' => 'DataValues\StringValue',
+					'unknown' => 'DataValues\UnknownValue',
+					'globecoordinate' => 'DataValues\Geo\Values\GlobeCoordinateValue',
+					'monolingualtext' => 'DataValues\MonolingualTextValue',
+					'multilingualtext' => 'DataValues\MultilingualTextValue',
+					'quantity' => 'DataValues\QuantityValue',
+					'time' => 'DataValues\TimeValue',
+					'wikibase-entityid' => 'Wikibase\DataModel\Entity\EntityIdValue',
+				)
+			),
+			new DataValueSerializer()
+		);
+		parent::__construct( null );
+	}
+
+	protected function configure() {
+		$defaultUser = $this->appConfig->offsetGet( 'defaults.user' );
+
+		$this
+			->setName( 'wm:wd:reference-as-statement-remove' )
+			->setDescription( 'Removes a statement whos property is only for references' )
+			->addOption(
+				'user',
+				null,
+				( $defaultUser === null ? InputOption::VALUE_REQUIRED :
+					InputOption::VALUE_OPTIONAL ),
+				'The configured user to use',
+				$defaultUser
+			)
+			->addOption(
+				'property',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Property to target'
+			);
+	}
+
+	protected function execute( InputInterface $input, OutputInterface $output ) {
+		// Get options
+		$user = $input->getOption( 'user' );
+		$userDetails = $this->appConfig->offsetGet( 'users.' . $user );
+		if ( $userDetails === null ) {
+			throw new RuntimeException( 'User not found in config' );
+		}
+
+		$propertyString = $input->getOption( 'property' );
+		$property = new PropertyId( $propertyString );
+		if ( $propertyString === null || $propertyString === '' || $property === null ) {
+			throw new RuntimeException( 'No property given' );
+		}
+
+		$output->writeln( 'Running SPARQL query to find items to check' );
+		$queryBuilder = new QueryBuilder( array(
+			'wdt' => 'http://www.wikidata.org/prop/direct/',
+		) );
+
+		$itemIds = $this->sparqlQueryRunner->getItemIdsFromQuery(
```

Comment:
Infact, right now if this uses spar it may have to go in the Wikipedia commands repo... As there is no way to generally deal with sparql stuff in addwiki stuff yet! :(